### PR TITLE
feat: generate valid passwords when using additional validators

### PR DESCRIPTION
**Pull request by johanseto**
## Description

(cherry picked from commit d21be1bb1a3d0171567b047442e136ebb8c01d3e)

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

install django-password-validators=1.7, set correct settings, add the package to installed apps and migrate it.
Peek 2024-01-22 15-01
[Screencast from 22-01-24 15:13:17.webm](https://github.com/nelc/edx-platform/assets/51926076/a9f351f5-d906-49de-9b51-7797fbdeec07)

### Before

[Screencast from 22-01-24 15:10:49.webm](https://github.com/nelc/edx-platform/assets/51926076/423c4f4a-7f79-4c8f-a3ff-266aa60d2c63)

## After
[Screencast from 22-01-24 15:15:17.webm](https://github.com/nelc/edx-platform/assets/51926076/1d52b6be-8014-4566-a552-e9231d8033a3)

## Other information

PRs related
https://github.com/edunext/edunext-platform/pull/676

### CODE REVIEW

File: openedx/core/djangoapps/user_api/management/commands/cancel_user_retirement_request.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 from openedx.core.djangoapps.user_api.accounts.utils import handle_retirement_cancellation
 from openedx.core.djangoapps.user_api.models import UserRetirementStatus
+from openedx.core.djangoapps.user_authn.utils import generate_password
```
Comment (andrey-canon) [inline openedx/core/djangoapps/user_api/management/commands/cancel_user_retirement_request.py:14]: is this necessary ?

**Review by andrey-canon — COMMENTED** [review]
what do you think about moving this feature outside this repository?

**Review by andrey-canon — COMMENTED** [review]
what do you think about moving this feature outside this repository?

File: openedx/core/djangoapps/user_api/management/commands/cancel_user_retirement_request.py (hunk 1 above)
Comment (johanseto) [reply openedx/core/djangoapps/user_api/management/commands/cancel_user_retirement_request.py]: no xD

**Comment by johanseto** [comment]
> what do you think about moving this feature outside this repository?

Do you think moving it to eox_nelp and somehow monkeypatch `generate_password` method with the eox-nelp one?.

Or do you think using a fork of edx_django_utils and change there `generate_password`??

**Comment by andrey-canon** [comment]
> > what do you think about moving this feature outside this repository?
> 
> @andrey-canon Do you think moving it to eox_nelp and somehow monkeypatch `generate_password` method with the eox-nelp one?.
> 
> Or do you think using a fork of edx_django_utils and change there `generate_password`??

I think both are valid options, probably the ideal one is to  fork edx_django_utils and the make then upstream pr, however I'm not sure  if this will be approve, so we could start with the monkey patch approach and then  migrate it

**Comment by johanseto** [comment]
Reaching the discussion, this PR is closed in favor of https://github.com/eduNEXT/eox-nelp/pull/130

**Pull request closed without merging**